exec/java-exec/src/main/java/org/apache/drill/exec/physical/impl/xsort/ExternalSortBatch.java

```diff
@@ -110,11 +111,12 @@
   private LinkedList<BatchGroup> batchGroups = Lists.newLinkedList();
   private LinkedList<BatchGroup> spilledBatchGroups = Lists.newLinkedList();
   private SelectionVector4 sv4;
-  private FileSystem fs;
+  private static FileSystem fs;
```

Comment:
Won't making this static and closing it cause other queries with an external sort to fail? You should be able to put this in the close() method of this class so that it only impacts the local instance. It looks like the directory created includes the operator ID, so there shouldn't be a need to keep around the spill directory until the foreman close method.